src/dagmc/DagMC.hpp

```diff
@@ -382,16 +387,25 @@ class DagMC {
 
   /** Get the instance of MOAB used by functions in this file. */
   Interface* moab_instance() {return MBI;}
+  std::shared_ptr<Interface> moab_instance_sptr() {
+    if (nullptr == MBI_shared_ptr)
+      std::runtime_error("MBI instance is not defined as a shared pointer !");
+    return MBI_shared_ptr;
+  }
```

Comment:
Added a runtime_error when trying to return the `MBI` instance as a `shared_ptr` when it has been provided as a direct pointer. (It has been implemented in the Header, as it is fairly simple, I can move it to the `cpp` file if you prefer)